ext/tiny_tds/client.c

```diff
@@ -171,7 +177,7 @@ static void rb_tinytds_client_reset_userdata(tinytds_client_userdata *userdata)
   userdata->dbsqlok_sent = 0;
   userdata->dbcancel_sent = 0;
   userdata->nonblocking = 0;
-  userdata->nonblocking_error.is_set = 0;
+  userdata->nonblocking_errors_length = 0;
```

Comment:
We can't free the memory here if we're freeing the memory in `nogvl_cleanup`. When this is called there is nothing to free. I am open to adding `userdata->nonblocking_errors_size = 0` here for consistency or going the other route and removing the reset for `userdata->nonblocking_errors_length`.

I decided on initializing/freeing the array in `nogvl_setup` and `nogvl_cleanup` because that marks the start and end of the time in which we need to make use of the errors array. So the array is around just long enough for its purpose. If we want the errors array to have a longer life cycle I'd be open to discussing that. For example, now that we are capturing multiple errors we may want to add a way to access those errors after we raise the exception (see #78). In that case it might be worth keeping the array around longer.

Reply:
>  I am open to adding userdata->nonblocking_errors_size = 0 here for consistency or going the other route and removing the reset for userdata->nonblocking_errors_length

I think the consistency would be preferable, as well as possibly a code comment as to why they're being zero'd but not free'd.

> I decided on initializing/freeing the array in nogvl_setup and nogvl_cleanup because that marks the start and end of the time in which we need to make use of the errors array. So the array is around just long enough for its purpose. 

Makes sense, thanks